core/src/queue.rs

```diff
@@ -335,6 +330,31 @@ impl Queue {
             });
     }
 
+    /// Overview:
+    /// 1. Transaction is added to queue using [`Queue::push`] method.
+    /// 2. Transaction is moved to [`Sumeragi::transaction_cache`] using [`Queue::pop_from_queue`] method.
+    ///    (transaction is removed from [`Queue::tx_hashes`], but kept in [`Queue::accepted_tx`])
+    /// 3. When transaction is removed from [`Sumeragi::transaction_cache`]
+    ///    (either because it was expired, or because transaction is commited to blockchain),
+    ///    we should remove transaction from [`Queue::accepted_tx`].
+    ///
+    /// `is_expired` is passed as a small optimization to not again call `self.is_expired(tx)`.
+    pub fn remove_stale_transaction(&self, tx: &AcceptedTransaction, is_expired: bool) {
+        let removed = self.accepted_txs.remove(&tx.as_ref().hash());
```

Comment:
why don't we remove transaction from `accepted_txs` at the time when we remove `tx_hashes`?

Reply:
It is used to return `Error::IsInQueue` when pushing same transaction twice

Reply:
neat, I didn't see that

Reply:
can you add a comment about that?